.github/workflows/snyk-test.yml

```diff
@@ -0,0 +1,59 @@
+name: Snyk Scan and Report
+
+on:
+  pull_request:
+    branches: [ main ]
+  schedule:
+    - cron:  '0 4 * * *'
+
+permissions:
+  id-token: write
+
+jobs:
+  snyk_run:
+    name: Snyk Run (for PR)
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
+    
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
```

Comment:
We currently use checkout@v3 on our application since that is the latest version.  Please update.